packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5711,23 +5711,25 @@ class RestorableRouteFuture<T> extends RestorableProperty<String?> {
   static NavigatorState _defaultNavigatorFinder(BuildContext context) => Navigator.of(context);
 }
 
-extension <T> on Iterable<T> {
-  T? firstWhereOrNull(bool Function(T element) test) {
-    for (final T element in this) {
-      if (test(element)) {
-        return element;
-      }
+/// First item satisfying the predicate, or null if not found.
+@visibleForTesting
+T? iterableFirstWhereOrNull<T>(Iterable<T> iterable, bool Function(T element) test) {
```

Comment:
make it private?

Reply:
Then IMHO we cannot test it in navigator_test.dart?

Reply:
IMHO then I cannot unit-test this function inside navigator_test.dart